### PR TITLE
WIP: feat: Geocoding glitch example

### PR DESCRIPTION
**Pull request by martinalbert**
## Description

Set up fastify with typescript and categorize code by profiles and usecases. I also moved superface directory into src because of typescript settings.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    "lint": "eslint src/",
+    "lint:fix": "eslint src/ --fix",
+    "format": "prettier -c src",
+    "format:fix": "prettier --write src/",
+    "start": "nodemon src/server.ts",
+    "prepush": "npm run build && npm run lint && npm run format",
+    "sf:install": "cd src && superface install"
+  },
+  "dependencies": {
+    "@superfaceai/cli": "^0.0.14",
```
Comment (Jakub-Vacek) [inline package.json:24]: use 0.0.16

File: src/pages/vcs.hbs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    </p>
+    
+    <form action="/vcs" method="post">
+      <label for="user">GitHub, GitLab or Bitbucket username</label><br>
+      <input type="text" id="user" name="user" value="superfaceai"><br>
+      <label for="service">Service</label><br>
+      <select name = "service" id="service">
+        <option value = "github" selected>GitHub</option>
+        <option value = "gitlab">GitLab</option>
+        <option value = "bitbucket">Bitbucket</option>
```
Comment (Jakub-Vacek) [inline src/pages/vcs.hbs:29]: Super.json uses only gitlab and github so Bitbucket should not probably be here.

File: package.json (hunk 1 above)
Comment (Jakub-Vacek) [reply package.json:24]: Wait, do you actually need CLI?

File: src/pages/vcs.hbs (hunk 2 above)
Comment (martinalbert) [reply src/pages/vcs.hbs:29]: I looked into super.json now and it seems that only `vcs/pull-request` profile is not using Bitbucket. This view is for listing user repos, so I guess it's ok.

**Comment by jnv** [comment]
TBH I kinda appreciated the original example for being short and sweet – all of the Superface-specific code is in a single file and there's not much more to review.
Have you considered splitting this more robust, more advanced example into a separate Glitch project?

**Comment by martinalbert** [comment]
I was aiming for good code browsing experience. I agree that it was simpler in original example, but as we add more use-cases, that one file will become big over time. I'm not sure how will that one big file affect developer experience, but if it would be guaranteed that developers find it short and sweet, then I'll gladly make it into one file again.

Do you think it would be better to have looser categorization (by profiles only), where one file would contain all use-cases for corresponding profile?

**Comment by jnv** [comment]
Maybe the question is what is the preferred outcome here? Is it having all capabilities in a single example app? Then yes, I think having single file per profile would be easier to grasp.

But personally I would try to keep things simple since this is a glitch project. It should be easy to get oriented and remix it. In particular, I recommend:

- Keeping one app per example, e.g. emailing repositories is one example, geocoding another one. 
- Treat TypeScript as secondary to pure JS examples. It's definitely worth to show it but it shouldn't be a prerequisite.
- Keep files structure simple and tooling to a necessary minimum. In my experience Glitch is best for quick and dirty prototyping and demoing, not for full fledged apps development, so in my opinion adding a draconian eslint config or unit tests isn't probably worth it.

Personally I think in the current form this would make for a really nice "kitchensink" example in create-superface-app or as a standalone repository.

**Review by zdne — COMMENTED** [review]
I am mostly missing the description of  what this example should be about - is this for the geocoding landing page? what usecases it should cover? 

As we discuss and also per @jnv comment  I'd stay with plain JS.

**Comment by martinalbert** [comment]
I'm closing this PR because of newly created repositories. This example(Geocode) now resides at https://github.com/superfaceai/geocode-example/. All of new repositories are in plain JS, but documentation is not complete yet. I'll let you know when they will be ready. 🙌🏻

**Comment by martinalbert** [comment]
@jnv @zdne documentation is complete, I'll remove this repository next week, so feel free to review other PRs
- [email example](https://github.com/superfaceai/email-example/pull/1)
- [geocode example](https://github.com/superfaceai/geocode-example/pull/1)
- [reverse geocode example](https://github.com/superfaceai/reverse-geocode-example/pull/1)

**Pull request closed without merging**